src/test/java/com/sap/oss/phosphor/fosstars/model/score/oss/ArtifactReleaseHistoryScoreTest.java

```diff
@@ -239,12 +243,154 @@ public void testFourteenMonthOldSingleVersion() {
     ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
     Value<ArtifactVersions> versions = RELEASED_ARTIFACT_VERSIONS.value(
         ArtifactVersions.of(version100));
-    ScoreValue value = score.calculate(versions);
-    Assert.assertEquals(0.0, value.get(), DELTA);
+    ScoreValue value = score.calculate(versions, ARTIFACT_VERSION.value(version100));
+    assertEquals(0.0, value.get(), DELTA);
   }
 
   @Test(expected = IllegalArgumentException.class)
   public void testWithNoInfo() {
     new ArtifactReleaseHistoryScore().calculate();
   }
-}
+
+  @Test
+  public void testVersionsSelectionByComparingMajor() {
+    LocalDateTime now = LocalDateTime.now();
+    ArtifactVersion version100 = new ArtifactVersion("1.0.0", now.minusDays(7 * 31));
+    ArtifactVersion version110 = new ArtifactVersion("1.1.0", now.minusDays(6 * 31));
+    ArtifactVersion version120 = new ArtifactVersion("1.2.0", now.minusDays(5 * 31));
+    ArtifactVersion version130 = new ArtifactVersion("1.3.0-MIGHTY", now.minusDays(4 * 31));
+    ArtifactVersion version140 = new ArtifactVersion("1.4.0", now.minusDays(3 * 31));
+    ArtifactVersion version150 = new ArtifactVersion("1.5.0", now.minusDays(64));
+    ArtifactVersion version200 = new ArtifactVersion("2.0.0", now.minusDays(31));
+    ArtifactVersions artifactVersions = ArtifactVersions.of(version100, version110, version120,
+        version130, version140, version150, version200);
+
+    ArtifactReleaseHistoryScore score = new ArtifactReleaseHistoryScore();
+    Value<ArtifactVersions> versions = RELEASED_ARTIFACT_VERSIONS.value(artifactVersions);
+
+    ScoreValue value100 = score.calculate(versions, ARTIFACT_VERSION.value(version100));
+    assertEquals(8.8, value100.get(), DELTA);
+
+    ScoreValue value120 = score.calculate(versions, ARTIFACT_VERSION.value(version120));
+    assertEquals(8.8, value120.get(), DELTA);
+
+    ScoreValue value150 = score.calculate(versions, ARTIFACT_VERSION.value(version150));
+    assertEquals(8.8, value150.get(), DELTA);
+
+    ScoreValue value130 = score.calculate(versions, ARTIFACT_VERSION.value(version130));
+    assertEquals(8.8, value130.get(), DELTA);
+
+    ScoreValue value200 = score.calculate(versions, ARTIFACT_VERSION.value(version200));
+    assertEquals(6.1, value200.get(), DELTA);
```

Comment:
@artem-smotrakov Yes, it is expected.
1. If a cluster is found the rating should be specifically higher. As the score considers only the associated versions.
2. If a cluster is not found, then every version is included. This should not have a comparatively good score.

Reply:
As mentioned by Sourabh in case of a new major version release (one single version) it falls back to check all versions.
This may lead to a lower score for the latest version compared to a score to a specific major version branch.
In the case that a project release new versions fast (e.g. new version every month) also the latest (single) major version would have a good score.
In cases that e.g. 1.x branch has slow release cycle, 2.x has fast cycle (both together has an average cycle) the latest 3.x version will have an average cycle. Compared to 1.x it is better, compared to 2.x is worse. But if they continue to release for 3.x fast it will be good.

Maybe this can be improved somehow, but currently I have no idea how to do this.

Reply:
Thanks for considering this. I am fine with either addressing it now or later. It is up to you, I've approved the current changeset.

Reply:
#569